### ISSUE TITLE
Quantile Regression Loss

### ISSUE THREAD

**Issue by vincentqb** [pytorch/pytorch#38035]
When interested in fitting an unknown probability distribution using a neural network model, we are sometimes interested in quantile regression. I suggest we add the quantile regression loss to the losses we have in pytorch. Example implementation [here](https://medium.com/the-artificial-impostor/quantile-regression-part-2-6fdbc26b2629).

**Comment by mruberry** [comment]
Great idea, @vincentqb!

**Comment by maxmarketit** [comment]
I have implemented `QuantileLoss` and `QinputLoss` in [this notebook](https://github.com/maxmarketit/Auto-PyTorch/blob/develop/examples/quantiles/Quantiles.ipynb).

 I followed the `pytorch` Loss Class and I do not know anything about `torch.jit.is_scripting()` part so it is all commented out. Anyone might help here. 

 It works fine in the example and I tested it with p(p>2) input features and even q(q>2) output features(for `QinputLoss`, and I figure I should have generalized the `QuantileLoss` too with tensor output like `batch size * output variables * quantiles`).

 Take a look at it and be free to say any suggestions.

https://github.com/maxmarketit/Auto-PyTorch/blob/develop/examples/quantiles/Quantiles.ipynb

**Comment by Javicadserres** [comment]
Hi, I don't know if there has been any update on this topic but I implemented the Quantile loss function [here](https://github.com/Javicadserres/wind-production-forecast/blob/28310d7dab7b47d7db3d690580505c1a456e471b/src/model/losses.py#L5) with Pytorch.

**Comment by EdisonAyranisthebest** [comment]
Hi! I’d like to take this on as an OSS contribution.
Proposal: add torch.nn.QuantileLoss (pinball loss) and torch.nn.functional.quantile_loss with quantile as float or 1D tensor, supporting reductions (none/mean/sum) and broadcasting.
I can include gradcheck + unit tests and docs.
Does this API naming/design match what maintainers want?
